corehq/apps/userreports/views.py

```diff
@@ -1348,17 +1348,21 @@ def rebuild_data_source(request, domain, config_id):
 
 def _number_of_records_to_be_iterated_for_rebuild(datasource_configuration):
     if datasource_configuration.referenced_doc_type == 'CommCareCase':
-        case_types = datasource_configuration.get_case_type_or_xmlns_filter()
-        # case_types could also be [None]
-        case_types = list(filter(None, case_types))
+        case_types = [
+            case_type
+            for case_type in datasource_configuration.get_case_type_or_xmlns_filter()
+            if case_type is not None
```

Comment:
just curious about this change, filter didn't work as expected?

Reply:
filter works fine. I changed it because (in my experience) comprehensions are more readable than filter, and by using a comprehension the comment became obvious and could be removed.

Reply:
Cool, thank you :+1: